server/src/main/java/io/littlehorse/server/streamsimpl/coreprocessors/CommandProcessor.java

```diff
@@ -28,22 +29,42 @@ public class CommandProcessor
     private KafkaStreamsServerImpl server;
     private final WfSpecCache wfSpecCache;
 
+    private final String storeName;
+
+    private final boolean isMetadataProcessorInstance;
+
     public CommandProcessor(
         LHConfig config,
         KafkaStreamsServerImpl server,
-        WfSpecCache wfSpecCache
+        WfSpecCache wfSpecCache,
+        String storeName,
+        boolean isMetadataProcessorInstance
     ) {
         this.config = config;
         this.server = server;
         this.wfSpecCache = wfSpecCache;
+        this.storeName = storeName;
+        this.isMetadataProcessorInstance = isMetadataProcessorInstance;
     }
 
     @Override
     public void init(final ProcessorContext<String, CommandProcessorOutput> ctx) {
         // temporary hack
 
         this.ctx = ctx;
-        dao = new KafkaStreamsLHDAOImpl(this.ctx, config, server, wfSpecCache);
+        final LHStoreWrapper localStore = new LHStoreWrapper(
+            ctx.getStateStore(storeName),
+            config
+        );
+        dao =
+            new KafkaStreamsLHDAOImpl(
+                this.ctx,
+                config,
+                server,
+                wfSpecCache,
+                localStore,
+                isMetadataProcessorInstance
+            );
```

Comment:
This is fine for now; however, in a follow-up PR let's add:
* `GlobalMetadataDAO`
* `ROGlobalMetaStore`

That should simplify the `KafkaStreamsLHDAOImpl`